lib/Doctrine/DBAL/DriverManager.php

```diff
@@ -259,10 +259,14 @@ private static function parseDatabaseUrl(array $params)
         }
         
         if (isset($url['path'])) {
+            $nameKey = isset($url['scheme']) && strpos($url['scheme'], 'sqlite') !== false
+                ? 'path' // pdo_sqlite driver uses 'path' instead of 'dbname' key
+                : 'dbname';
+
             if (!isset($url['scheme']) || (strpos($url['scheme'], 'sqlite') !== false && $url['path'] == ':memory:')) {
```

Comment:
Can we somehow try to avoid duplicate checks on whether the path is a sqlite path here?
Also I think it would be a good idea to consider setting `memory` parameter instead of `path` parameter, if the URL path refers to in-memory (it makes no difference for the driver but could be useful in userland when checking for specific parameters IMO).

Otherwise this patch looks good to me.